service/controller/app/resource/chartoperator/resource.go

```diff
@@ -140,6 +149,79 @@ func (r Resource) installChartOperator(ctx context.Context, cr v1alpha1.App) err
 	return nil
 }
 
+func (r Resource) triggerReconciliation(ctx context.Context, operatorApp v1alpha1.App) error {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Find all the Apps CR for a given cluster.
+	// If `operatorApp` is an org-namespaced App use the cluster label selector.
+	var appList v1alpha1.AppList
+	{
+		o := client.ListOptions{}
+
+		var selector k8slabels.Selector
+
+		if key.IsInOrgNamespace(operatorApp) {
+			selector, err = k8slabels.Parse(fmt.Sprintf("%s=%s", label.Cluster, key.ClusterLabel(operatorApp)))
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			o.LabelSelector = selector
+		}
+
+		err = r.ctrlClient.List(ctx, &appList, &o)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	// For each App, check if the corresponding Chart CR exists.
+	// If not, annotate the App to trigger the reconciliation.
+	for i, app := range appList.Items {
+		// Skip for in-cluster apps and the chart-operator app itself.
+		if key.InCluster(app) || app.ObjectMeta.Name == operatorApp.ObjectMeta.Name {
+			continue
+		}
+
+		name := key.ChartName(app, r.workloadClusterID)
+
+		var chart v1alpha1.Chart
+		err = cc.Clients.K8s.CtrlClient().Get(
+			ctx,
+			types.NamespacedName{Name: name, Namespace: r.chartNamespace},
+			&chart,
+		)
+
+		// if chart CR is not found, trigger sync
+		if apierrors.IsNotFound(err) {
+			r.logger.Debugf(ctx, "did not find chart %#q in namespace %#q", name, r.chartNamespace)
+			r.logger.Debugf(ctx, "annotating %#q app", app.ObjectMeta.Name)
+
+			if len(app.GetAnnotations()) == 0 {
+				app.Annotations = map[string]string{}
+			}
+
+			modifiedApp := app.DeepCopy()
+			modifiedApp.Annotations[AppOperatorTriggerReconciliation] = metav1.Now().Format(time.RFC3339)
+
+			// Using indexing to fix the `G601: Implicit memory aliasing in for loop.`
+			err = r.ctrlClient.Patch(ctx, modifiedApp, client.MergeFrom(&appList.Items[i]))
```

Comment:
Is it maybe easier to generate the patch directly? instead of generating the patch from a deepcopy. I believe submitting the patch directly (as e.g. a JSONPatch) would guard us against possible race conditions.

Reply:
Thanks for the suggestion @MarcelMue ! I will give it a try

Reply:
@MarcelMue fixed 🙏